builder/build_test.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) OpenFaaS Project 2017. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package builder
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"testing"
+)
+
+func Test_CopyFiles(t *testing.T) {
```

Comment:
@itscaro thoughts on these tests?

Reply:
@ganesshkumar IMO, it's more relevant to assert file modes for ops done by `cp` function, which is in charge of doing the stuff with file modes. The test for `CopyFiles` should test only the number of files copied and the name of the destination dir.

The current test will fail on Windows, you can rename `build_test` to `build_unix_test` and add this build flag `// +build darwin dragonfly freebsd linux netbsd openbsd solaris` at the beginning of the test file

Example:
https://github.com/openfaas/faas-cli/compare/master...itscaro:pr/254?expand=1